docs/en/engines/database-engines/overlay.md

```diff
@@ -0,0 +1,130 @@
+- `DatabaseFilesystem`: allows to read-only interact with files stored on the file system. Internally, it uses `TableFunctionFile` to implicitly load file when a user requests the table. Result of `TableFunctionFile` call cached inside to provide quick access.
+- `DatabaseS3`: allows to read-only interact with s3 storage. It uses `TableFunctionS3` to implicitly load table from s3
+`DatabaseHDFS`: allows to interact with hdfs storage. It uses `TableFunctionHDFS` to implicitly load table from hdfs
+
+---
+
+## What operations does the overlay support? {#implementation}
+
+### Table/DB discovery {#discovery}
+
+* `SHOW TABLES FROM dboverlay` — returns the **union** of member DB tables.
+* `SELECT … FROM dboverlay.table` — **reads** transparently hit the underlying table.
+
+### Mutating operations (facade mode) {#operations}
+
+| Operation                  | Behavior                                                                                  |
+| :------------------------- | :---------------------------------------------------------------------------------------- |
+| `CREATE TABLE dboverlay.*` | **Rejected** — throws `BAD_ARGUMENTS` with a clear message to create in an underlying DB. |
+| `ATTACH TABLE dboverlay.*` | **Rejected** — `BAD_ARGUMENTS`.                                                           |
+| `ALTER TABLE dboverlay.*`  | **Pass-through** — forwards to the appropriate underlying DB.                             |
+| `RENAME TABLE dboverlay.*` | **Rejected** — `BAD_ARGUMENTS`.                                                           |
+| `DROP TABLE dboverlay.*`   | **No-op** — ignored (allows `DROP DATABASE dboverlay` to succeed).                        |
+| `INSERT INTO dboverlay.*`  | **Pass-through** — executes against the table in the corresponding underlying DB.         |
+
+> Rationale: the facade is a **view**. Data-definition & data-mutation happen in the member databases.
+
+### Database life cycle (facade mode) {#life-cycle}
+
+* `DROP DATABASE dboverlay` — **succeeds** and **does not** cascade to members.
+
+  * Implementation details that make this safe:
+
+    * `getUUID()` returns **Nil** (no DB UUID mapping).
+    * `getTablesIterator(ctx, filter, /*skip_not_loaded=*/true)` returns **empty** (so the dropper doesn’t see tables).
+    * `empty()` returns **true** (facade reports empty during destructive passes).
+    * `dropTable()` is a **no-op** (dropper won’t fail mid-way).
+    * `drop()` is a **no-op** (doesn’t touch member DBs).
+* `isReadOnly()` returns **true** in case of server (**readonly**) implementation (so the server doesn’t block `DROP DATABASE` up front). Mutating table ops are still guarded individually.
+
+---
+
+## Resilience & safety {#safty}
+
+* **Null-safety**: all loops over `databases` skip `nullptr` entries; `registerNextDatabase()` rejects `nullptr`.
+  This prevents crashes from bad metadata or attach order.
```

Comment:
I think it is OK to just remove this paragraph, since 

>all loops over `databases` skip `nullptr` entries

this looks like outdated, there is no null checks and should not be (as I commented above)